### PR TITLE
refactor: Migrate AdhocFilterEditPopoverSqlTabContent to TypeScript

### PR DESCRIPTION
**Pull request by EnxDev**
### SUMMARY
Migrate AdhocFilterEditPopoverSqlTabContent to TypeScript

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
N.A.

### TESTING INSTRUCTIONS
All tests should pass

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by michael-s-molina** [comment]
@EnxDev

File: superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterEditPopoverSqlTabContent/index.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      options.filter(
+        (option): option is ColumnMeta =>
+          typeof option === 'object' &&
+          option !== null &&
+          'column_name' in option &&
+          typeof option.column_name === 'string' &&
+          'type' in option,
+      ),
+    ),
+  );
```
Comment (justinpark) [inline superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterEditPopoverSqlTabContent/index.tsx:89]: Please wrap this block by useMemo for the performance.

```suggestion
  const keywords = useMemo(() => sqlKeywords.concat(
    getColumnKeywords(
      options.filter(
        (option): option is ColumnMeta =>
          typeof option === 'object' &&
          option !== null &&
          'column_name' in option &&
          typeof option.column_name === 'string' &&
          'type' in option,
      ),
    ),
  ), [sqlKeywords]);
```

File: superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterEditPopoverSqlTabContent/index.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+  const selectOptions = Object.values(Clauses).map(clause => ({
+    label: clause,
+    value: clause,
+  }));
```
Comment (justinpark) [inline superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterEditPopoverSqlTabContent/index.tsx:94]: please wrap this block by useMemo for the performance.

```suggestion
  const selectOptions = useMemo(() => Object.values(Clauses).map(clause => ({
    label: clause,
    value: clause,
  })), [Clauses]);
```

File: superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterEditPopoverSqlTabContent/index.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  onChange: (filter: AdhocFilter) => void;
+  options: OptionSortType[];
+  height: number;
+}) {
+  const aceEditorRef = useRef(null);
+  const theme = useTheme();
+
+  const handleAceEditorRef = useCallback(ref => {
+    if (ref) aceEditorRef.current = ref;
+  }, []);
```
Comment (justinpark) [inline superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterEditPopoverSqlTabContent/index.tsx:53]: it's unnecessary. You can directly bind `aceEditorRef` like `ref={aceEditorRef}`

Comment (EnxDev) [reply superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterEditPopoverSqlTabContent/index.tsx:53]: You're absolutely right. Binding aceEditorRef directly with ref={aceEditorRef} is sufficient, as it doesn't create new functions on every render. Using useCallback for this scenario is unnecessary and adds extra complexity without any practical advantage. Thank you for pointing that out! :)

**Review by geido — APPROVED** [review]

**Pull request merged**